### PR TITLE
Implement more WinUI SwitchHandler properties

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Implement more WinUI SwitchHandler properties:
- TrackColor
- ThumbColor

### PR Checklist ###

- [x] Targets the correct branch 
- [ ] Tests are passing (or failures are unrelated)
- [x] Adds the property to the appropriate interface
- [x] Avoids any changes not essential to the handler property
- [x] Adds the mapping to the PropertyMapper in the handler
- [x] Adds the mapping method to the Android, iOS, and Standard aspects of the handler
- [x] Implements the actual property updates (usually in extension methods in the Platform section of Core)
- [x] Tags ported renderer methods with [PortHandler]
- [x] Adds an example of the property to the sample project (MainPage)
- [x] Adds the property to the stub class
- [x] Implements basic property tests in DeviceTests

#### Does this PR touch anything that might effect accessibility?
No

### CODE REVIEW

**Review by rmarinho — CHANGES_REQUESTED** [review]
Looking at this sample i don't see the ThumbColor changing is this expected ? 

`verticalStack.Add(new Switch() { OnColor = Colors.Green,  ThumbColor = Colors.Yellow });`

image

image

**Comment by rmarinho** [comment]
It's  working better, but seems first time it still shows the default color, and the  `TrackColor` just changes  when it loses focus:
image


image

**Review by rachelkang — CHANGES_REQUESTED** [review]
seconding Rui - the track color doesn't initialize properly

**Comment by jsuarezruiz** [comment]
Could we get another review for the PR again? The Thumb color is set correctly regardless of state (IsToggled). As for OnColor, if the state is Off, the color is not set until the state changes, if it is On, the track background color is applied directly.

**Comment by rachelkang** [comment]
> Could we get another review for the PR again? The Thumb color is set correctly regardless of state (IsToggled). As for OnColor, if the state is Off, the color is not set until the state changes, if it is On, the track background color is applied directly.

@jsuarezruiz the issue still remains with the OnColor switch - the first time the switch is toggled on, the track color turns blue. Then, it automatically changes to red and remains red. Whenever the switch is toggled on after that, it goes straight to red

**Review by rmarinho — APPROVED** [review]

File: src/Core/src/Platform/ImageSourceExtensions.cs [hunk 1]
```diff
@@ -32,6 +32,8 @@ static async Task LoadImageResult(Task<IImageSourceServiceResult<NativeImage>?>
 
 		public static Task<IImageSourceServiceResult<NativeImage>?> GetNativeImageAsync(this IImageSource imageSource, IMauiContext mauiContext)
 		{
+			if (imageSource == null)
+				return new Task<IImageSourceServiceResult<NativeImage>?>(() => null);
```
Comment (rmarinho) [inline src/Core/src/Platform/ImageSourceExtensions.cs:36]: @PureWeen  same NRE, this a better fix ?

**Review by rachelkang — COMMENTED** [review]
Previous issue I noted was fixed! Yay

One thing I'm still noticing though is that on focus/hover, the switch track color appears lighter - but if I'm seeing this correctly, this is only the case for the default blue ones. If we could somehow standardize and maintain the highlight on the custom track colors too (the red one here), I think that would be a good idea

winui_maui_switch_color

**Comment by rmarinho** [comment]
Created a issue for this @rachelkang  https://github.com/dotnet/maui/issues/2685

**Pull request merged**